### PR TITLE
fix no default samesite

### PR DESCRIPTION
**Pull request by bharat-rajani**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Go Version Update
- [ ] Dependency Update

## Description

This PR sets the SameSite cookie attribute to Lax in the Set-Cookie header. The SameSite=Lax value provides a reasonable balance between security and usability for websites.

Reference: 
https://owasp.org/www-community/SameSite

## Related Tickets & Documents

- Related Issue # https://github.com/gorilla/sessions/issues/256
- Closes # https://github.com/gorilla/sessions/issues/256

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Run verifications and test

- [X] `make verify` is passing
- [X] `make test` is passing

Fixes: https://github.com/gorilla/sessions/issues/256

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
 coverage.coverprofile
+
+# IDE Metadata
+.idea
+.vscode
```
Comment (jaitaiwan) [inline .gitignore:5]: I'd like to see these in a separate pull request.

**Review by jaitaiwan — CHANGES_REQUESTED** [review]
From what I read of your linked comments, it seems like we should set the default to "None" for backwards compatible behaviour.

**Comment by bharat-rajani** [comment]
@jaitaiwan https://github.com/gorilla/sessions/pull/276#pullrequestreview-2014240474
Some browsers are mandating additional Secure attribute when SameSite=None. 
( [chrome](https://developers.google.com/search/blog/2020/01/get-ready-for-new-samesitenone-secure#:~:text=When%20the%20SameSite%3DNone%20attribute,provide%20protection%20against%20network%20attacks.), [firefox](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#none) ) 

I think it will be good to add Secure along with SameSite=None, let me know what you think!

**Comment by jaitaiwan** [comment]
I think that's a good idea. We can release it under a major version tag so it doesn't break folk's installations either way.

**Comment by jaitaiwan** [comment]
LGTM - Just waiting for scanning/tests to finish

**Comment by bharat-rajani** [comment]
@jaitaiwan I believe vulncheck issues are unrelated and we can tackle them separately. 
Also, I ran linter locally and I don't see issues which are somehow reported in github workflow.
```
docker run -t --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.53.3 golangci-lint run -v
```

Can you help?

**Comment by apoorvajagtap** [comment]
The linter is passing for me as well. I think this could be a false positive, I can't even find the existence of `max` reported in the error. 
I think we can ignore the lint errors, and as @bharat-rajani mentioned we can work through the vuln failures seperately. @jaitaiwan WDYT?

**Comment by jaitaiwan** [comment]
We'll merge this once we've fixed #277 I think

**Comment by jaitaiwan** [comment]
I think this might be missing from v1.4.0 can you check @bharat-rajani or @apoorvajagtap ?

**Comment by yhlee-tw** [comment]
@jaitaiwan the new default `None`, like what @bharat-rajani has mentioned, breaks setup where they disable Secure (e.g. local HTTP server) . it's an easy fix but I thought it was planned for a major version bump (or at least, mentioned in release notes)

**Pull request merged**